actors/builtin/miner/miner_actor.go

```diff
@@ -2067,6 +2068,288 @@ func (a Actor) RepayDebt(rt Runtime, _ *abi.EmptyValue) *abi.EmptyValue {
+
+				newSectorInfo := updateWithDetails.sectorInfo
+
+				newSectorInfo.SealedCID = updateWithDetails.update.NewSealedSectorCID
+				newSectorInfo.DealIDs = updateWithDetails.update.Deals
+
+				newSectorInfo.DealWeight = updateWithDetails.dealWeight.DealWeight
+				newSectorInfo.VerifiedDealWeight = updateWithDetails.dealWeight.VerifiedDealWeight
+
+				// compute initial pledge
+				duration := updateWithDetails.sectorInfo.Expiration - rt.CurrEpoch()
+
+				// TODO: Should we include this?
+				//if duration < MinSectorExpiration {
+				//	rt.Log(rtt.WARN, "precommit %d has lifetime %d less than minimum. ignoring", precommit.Info.SectorNumber, duration, MinSectorExpiration)
+				//	continue
+				//}
+
+				pwr := QAPowerForWeight(info.SectorSize, duration, newSectorInfo.DealWeight, newSectorInfo.VerifiedDealWeight)
+
+				newSectorInfo.ReplacedDayReward = updateWithDetails.sectorInfo.ExpectedDayReward
+				newSectorInfo.ExpectedDayReward = ExpectedRewardForPower(rewRet.ThisEpochRewardSmoothed, powRet.QualityAdjPowerSmoothed, pwr, builtin.EpochsInDay)
+				newSectorInfo.ExpectedStoragePledge = ExpectedRewardForPower(rewRet.ThisEpochRewardSmoothed, powRet.QualityAdjPowerSmoothed, pwr, InitialPledgeProjectionPeriod)
+				newSectorInfo.ReplacedSectorAge = maxEpoch(0, rt.CurrEpoch()-updateWithDetails.sectorInfo.Activation)
+
+				initialPledgeAtUpgrade := InitialPledgeForPower(pwr, rewRet.ThisEpochBaselinePower, rewRet.ThisEpochRewardSmoothed,
+					powRet.QualityAdjPowerSmoothed, rt.TotalFilCircSupply())
+
+				if initialPledgeAtUpgrade.GreaterThan(updateWithDetails.sectorInfo.InitialPledge) {
+					deficit := big.Sub(initialPledgeAtUpgrade, updateWithDetails.sectorInfo.InitialPledge)
+
+					unlockedBalance, err := st.GetUnlockedBalance(rt.CurrentBalance())
+					builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to calculate unlocked balance")
+					builtin.RequirePredicate(rt, unlockedBalance.GreaterThanEqual(deficit), exitcode.ErrInsufficientFunds, "insufficient funds for new initial pledge requirement %s, available: %s, skipping sector %d",
+						deficit, unlockedBalance, updateWithDetails.sectorInfo.SectorNumber)
+
+					err = st.AddInitialPledge(deficit)
+					builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to add initial pledge")
+
+					newSectorInfo.InitialPledge = initialPledgeAtUpgrade
+				}
+
+				var partition Partition
+				found, err := partitions.Get(updateWithDetails.update.Partition, &partition)
+
+				builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load deadline %v partition %v",
+					updateWithDetails.update.Deadline, updateWithDetails.update.Partition)
+
+				if !found {
+					rt.Abortf(exitcode.ErrNotFound, "no such deadline %v partition %v", dlIdx, updateWithDetails.update.Partition)
+				}
+
+				// TODO: Group by partition
```

Comment:
I've been a keener since middle school, so I wanna do this too, but it'll be a future thing. 

or a never thing.